x-pack/apm-server/sampling/eventstorage/pebble.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package eventstorage
+
+import (
+	"bytes"
+	"path/filepath"
+
+	"github.com/cockroachdb/pebble/v2"
+	"github.com/cockroachdb/pebble/v2/bloom"
+
+	"github.com/elastic/apm-server/internal/logs"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+func eventComparer() *pebble.Comparer {
+	comparer := *pebble.DefaultComparer
+	// Required for prefix bloom filter
+	comparer.Split = func(k []byte) int {
+		return bytes.IndexByte(k, ':')
+	}
+	return &comparer
+}
+
+func OpenEventPebble(storageDir string) (*pebble.DB, error) {
+	opts := &pebble.Options{
+		FormatMajorVersion: pebble.FormatColumnarBlocks,
+		Logger:             logp.NewLogger(logs.Sampling),
+		MemTableSize:       16 << 20,
+		Levels: []pebble.LevelOptions{
+			{
+				BlockSize:    16 << 10,
+				Compression:  func() pebble.Compression { return pebble.SnappyCompression },
+				FilterPolicy: bloom.FilterPolicy(10),
+				FilterType:   pebble.TableFilter,
+			},
+		},
+		Comparer: eventComparer(),
+	}
+	opts.Experimental.MaxWriterConcurrency = 1 // >0 enables parallel writers, the actual value doesn't matter
+	return pebble.Open(filepath.Join(storageDir, "event"), opts)
+}
+
+func OpenDecisionPebble(storageDir string) (*pebble.DB, error) {
+	return pebble.Open(filepath.Join(storageDir, "decision"), &pebble.Options{
+		FormatMajorVersion: pebble.FormatColumnarBlocks,
+		Logger:             logp.NewLogger(logs.Sampling),
+		MemTableSize:       2 << 20,
+		Levels: []pebble.LevelOptions{
+			{
+				BlockSize:    2 << 10,
+				Compression:  func() pebble.Compression { return pebble.NoCompression },
+				FilterPolicy: bloom.FilterPolicy(10),
+				FilterType:   pebble.TableFilter,
+			},
+		},
+	})
+}
```

Comment:
[to reviewer] would love feedback from folks with RocksDB configuration expertise. I obtained these config options by trial and error and isn't super scientific. TBH I'm rather new to this.

Reply:
As mentioned we are now conducting experiments.